programs/tests/mt-c-tests/uart/uart.c

```diff
@@ -0,0 +1,236 @@
+/**
+ * @author Magnus Mæhlum (magnmaeh@stud.ntnu.no)
+ * @brief This test implements a Software-defined UART, which is a more formal
+ *        name for a bit-banged UART.
+ * 
+ */
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <errno.h>
+#include <string.h>
+#include <flexpret.h>
+
+#define CLOCKFREQ     ((int)(100e6)) // 100 MHz
+#define NS_PER_CLK    (10)
+#define UART_BAUDRATE (38400)
+#define CLKS_PER_BAUD ((int) ((CLOCKFREQ) / (UART_BAUDRATE)))
+#define NS_PER_BAUD   ((CLKS_PER_BAUD) * (NS_PER_CLK))
+
+enum State {
+    STATE_STARTBIT,
+    STATE_DATABITS,
+    STATE_STOPBIT,
+} state = STATE_STARTBIT;
+
+struct RingBuffer {
```

Comment:
A ring-buffer is also something that would be nice to have as a lib that can be optionally included by users